src/client/Microsoft.Identity.Client/Region/IRegionDiscoveryProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Region
+{
+    internal interface IRegionDiscoveryProvider
```

Comment:
This provider does not behave like the other providers and it does too little, and so you are moving a lot of the logic to the manager. (In hindsight, I should have created an IProvider interface to enforce this...)

Please have a look at `NetworkMetadataProvider`, as the region provider is almost identical in functionality to it.

- does some stuff (env variable / network call) too figure out the region
- once you figure out the region, inject it into the static cache
- return an `InstanceDiscoveryMetadataEntry`

Reply:
After looking at the code in more detail, I think your solution is simpler and it is ok,